src/connectors/openrouter.py

```
@@ -69,26 +85,57 @@ def _resolve_headers_from_provider(self) -> dict[str, str]:
         provider = self.headers_provider
         errors: list[Exception] = []
 
-        if self.key_name is not None:
+        def _try_provider_call(*args: Any) -> dict[str, str] | None:
             try:
-                return provider(self.key_name, self.api_key)
+                result = provider(*args)
             except (AttributeError, TypeError) as exc:
                 errors.append(exc)
+                return None
+            except Exception as exc:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)